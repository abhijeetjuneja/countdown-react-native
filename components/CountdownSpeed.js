import React, { Component } from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';

export default class CountdownSpeed extends Component {
    render() {
        let { countdown, speed, start} = this.props
        console.log(speed)
        return (
            <View style={styles.container}>
                <Text>
                    <TouchableOpacity
                        style={countdown <= 0 || !start ? styles.disabledButton : speed == 1 ? styles.buttonStyle1 : styles.buttonStyle2}
                        onPress={() => this.props.setStates('speed',1)}
                        disabled={countdown <= 0 || !start}
                    ><Text style={countdown <= 0 || !start ? styles.disabledButtonText : speed == 1 ? styles.buttonText1 : styles.buttonText2}>1x</Text></TouchableOpacity>
                    <TouchableOpacity
                        style={countdown <= 0 || !start ? styles.disabledButton : speed == 1.5 ? styles.buttonStyle1 : styles.buttonStyle2}
                        onPress={() => this.props.setStates('speed',1.5)}
                        disabled={countdown <= 0 || !start}
                    ><Text style={countdown <= 0 || !start ? styles.disabledButtonText : speed == 1.5 ? styles.buttonText1 : styles.buttonText2}>1.5x</Text></TouchableOpacity>
                    <TouchableOpacity
                        style={countdown <= 0 || !start ? styles.disabledButton : speed == 2 ? styles.buttonStyle1 : styles.buttonStyle2}
                        onPress={() => this.props.setStates('speed',2)}
                        disabled={countdown <= 0 || !start}
                    ><Text style={countdown <= 0 || !start ? styles.disabledButtonText : speed == 2 ? styles.buttonText1 : styles.buttonText2}>2x</Text></TouchableOpacity>
                </Text>
            </View>
        )
    }
}


const styles = StyleSheet.create({
    container:{
        padding: '100',
        marginTop:50,
        alignItems: 'center',
        justifyContent: 'center',
    },
    disabledButton:{
        height: 50,
        width:100,
        marginTop:10,
        paddingTop:15,
        paddingBottom:15,
        marginLeft:10,
        backgroundColor:'#C0C0C0',
        borderRadius:10,
        borderWidth: 1,
        borderColor: 'grey',
        color:'grey',
        shadowColor: 'rgba(0,0,0, .4)', // IOS
        shadowOffset: { height: 1, width: 1 }, // IOS
        shadowOpacity: 1, // IOS
        shadowRadius: 1, //IOS
        elevation: 2, // Android
    },
    buttonStyle1:{
        height: 50,
        marginTop:10,
        width:100,
        paddingTop:15,
        paddingBottom:15,
        marginLeft:10,
        backgroundColor:'white',
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#000',
        shadowColor: 'rgba(0,0,0, .4)', // IOS
        shadowOffset: { height: 1, width: 1 }, // IOS
        shadowOpacity: 1, // IOS
        shadowRadius: 1, //IOS
        elevation: 2, // Android
    },
    buttonStyle2:{
        height: 50,
        marginTop:10,
        width:100,
        paddingTop:15,
        paddingBottom:15,
        marginLeft:10,
        backgroundColor:'black',
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#000',
        shadowColor: 'rgba(0,0,0, .4)', // IOS
        shadowOffset: { height: 1, width: 1 }, // IOS
        shadowOpacity: 1, // IOS
        shadowRadius: 1, //IOS
        elevation: 2, // Android
    },
    buttonText1: {
        color:'black',
        fontSize:'16px',
        fontWeight:500,
        alignContent:'center',
        alignSelf: 'center'
    },
    buttonText2: {
        color:'white',
        fontSize:'16px',
        fontWeight:500,
        alignContent:'center',
        alignSelf: 'center'
    },
    disabledButtonText: {
        color:'grey',
        fontSize:'16px',
        fontWeight:500,
        alignContent:'center',
        alignSelf: 'center'
    },
})