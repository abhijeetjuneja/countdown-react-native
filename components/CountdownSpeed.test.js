import React from 'react';
import CountdownSpeed from './CountdownSpeed';

import renderer from 'react-test-renderer';

test('countdownspeed renders correctly with start-false speed-1 pause-false', () => {
    const tree = renderer.create(<CountdownSpeed start={false} speed={1} pause={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


test('countdownspeed renders correctly with start-true speed-1 pause-false', () => {
    const tree = renderer.create(<CountdownSpeed start={true} speed={1} pause={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


test('countdownspeed renders correctly with start-true speed-1 pause-true', () => {
    const tree = renderer.create(<CountdownSpeed start={true} speed={1} pause={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


test('countdownspeed renders correctly with start-true speed-1.5 pause-true', () => {
    const tree = renderer.create(<CountdownSpeed start={true} speed={1.5} pause={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


test('countdownspeed renders correctly with start-true speed-2 pause-true', () => {
    const tree = renderer.create(<CountdownSpeed start={true} speed={2} pause={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('countdownspeed renders correctly with start-true speed-1.5 pause-false', () => {
    const tree = renderer.create(<CountdownSpeed start={true} speed={1.5} pause={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});


test('countdownspeed renders correctly with start-true speed-2 pause-false', () => {
    const tree = renderer.create(<CountdownSpeed start={true} speed={2} pause={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});