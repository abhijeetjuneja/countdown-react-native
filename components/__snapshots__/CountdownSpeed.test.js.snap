// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countdownspeed renders correctly with start-false speed-1 pause-false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#C0C0C0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "grey",
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#C0C0C0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "grey",
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1.5x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#C0C0C0",
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "grey",
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        2x
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdownspeed renders correctly with start-true speed-1 pause-false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "black",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1.5x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        2x
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdownspeed renders correctly with start-true speed-1 pause-true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "black",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1.5x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        2x
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdownspeed renders correctly with start-true speed-1.5 pause-false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "black",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1.5x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        2x
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdownspeed renders correctly with start-true speed-1.5 pause-true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "black",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1.5x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        2x
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdownspeed renders correctly with start-true speed-2 pause-false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1.5x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "black",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        2x
      </Text>
    </View>
  </Text>
</View>
`;

exports[`countdownspeed renders correctly with start-true speed-2 pause-true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 50,
    }
  }
>
  <Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        1.5x
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginLeft": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingTop": 15,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "black",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        2x
      </Text>
    </View>
  </Text>
</View>
`;
